projects/saturn/src/main/java/io/fairspace/saturn/webdav/DavFactory.java

```diff
@@ -60,6 +60,9 @@ public Resource getResource(org.apache.jena.rdf.model.Resource subject) {
     }
 
     public Resource getResource(org.apache.jena.rdf.model.Resource subject, Access access) {
+        if (isExtraStoreResource() && subject.hasProperty(RDF.type, FS.File) && !subject.hasProperty(FS.createdBy, currentUserResource())) {
```

Comment:
Having 'subject.hasProperty(RDF.type, FS.File)' means that a non-file resource (directory, collection) can be accessed by everyone.

Reply:
yes, I though for now there will be simple file-based permissions , as this is what we need for jupyter. In the future we can make extra storage permissions more complex (e.g. when we find another usecase for this storage).